extensions/resource-deployment/src/ui/deploymentDialog.ts

```diff
@@ -0,0 +1,135 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as azdata from 'azdata';
+import * as nls from 'vscode-nls';
+import * as vscode from 'vscode';
+import { DialogBase } from './dialogBase';
+import { INotebookService } from '../services/notebookService';
+import { DeploymentProvider, DialogFieldInfo, FieldType } from '../interfaces';
+
+const localize = nls.loadMessageBundle();
+
+export class DeploymentDialog extends DialogBase {
+
+	private variables: { [s: string]: string | undefined; } = {};
+
+	constructor(context: vscode.ExtensionContext,
+		private notebookService: INotebookService,
+		private deploymentProvider: DeploymentProvider) {
+		super(context, deploymentProvider.dialog.title, deploymentProvider.dialog.name, false);
+		this._dialogObject.okButton.label = localize('deploymentDialog.OKButtonText', 'Open Notebook');
+		this._dialogObject.okButton.onClick(() => this.onComplete());
+	}
+
+	protected initializeDialog() {
+		const tabs: azdata.window.DialogTab[] = [];
+		this.deploymentProvider.dialog.tabs.forEach(tabInfo => {
+			const tab = azdata.window.createTab(tabInfo.title);
+			tab.registerContent((view: azdata.ModelView) => {
+				const sections: azdata.FormComponentGroup[] = [];
+				tabInfo.sections.forEach(sectionInfo => {
+					const fields: azdata.FormComponent[] = [];
```

Comment:
Do we need any error handling for malformed sections? is that possible?

Reply:
it is possible, but it will be development time only, we are not reading user supplied content here, so I am not worried, we can add a schema validator later.